### ISSUE TITLE
About component (#94) to store version it was created with

### ISSUE THREAD

**Issue by xRadne** [strusoft/femdesign-api#189]
**About component** (#94) to store info of **what version of the API it was created with** and compare to the current installed. 

This would make it easier to find what version was used when creating an old script. This will probably be more important the more users and major versions (20, 21...) of the API we get.

_Originally posted by @xRadne in https://github.com/strusoft/femdesign-api/issues/188#issuecomment-1027832036_

**Comment by andosca** [comment]
Not really sure what you mean. If I would open your script created with 20.0.0, where you dumped an Info component, in 21.0.0 - what should the Info component show?

**Comment by xRadne** [comment]
**Both!**

**Comment by andosca** [comment]
How? :)

**Comment by xRadne** [comment]
Like this?

**Comment by xRadne** [comment]
> Maybe mark About component as Obsolete every new milestone?

I think this is a bad Idea. Extra work for us and the script author that needs to update the component. Also the OLD tag looks ugly 😉

I think we instead should list both version installed and version created on the component. 

> Example: 
> Created with: `20.0.0` 
> Currently installed: `21.2.1`

**Comment by andosca** [comment]
I don't know how to do it. Would be happy if you could explain how to implement this technically 😃

**Comment by xRadne** [comment]
> if you want to (de)serialize data to a gh file beyond the standard, you'll need to override the Write() and Read() methods.

https://www.grasshopper3d.com/forum/topics/how-to-save-additional-state-flag-of-my-custom-component

TLDR: Override serialization of the component. Ie, override methods `GH_Component.Read` and `GH_Component.Write`

**Comment by andosca** [comment]
Thanks

**Comment by Marco-Pellegrino** [comment]
Just a thought:

Should the user ALWAYS keep his library up to date?
What's the main reason to try to catch error/bug that happen in an olver release?

Should we force the user to install the new release?
As user, I was always replacing the "old component" with the most newer when I was encounter issues.

Do we want to give support to user with a not up-to-date version?

**Comment by andosca** [comment]
@Marco-Pellegrino 

Say you have invested a lot of time creating different scripts to automate certain processes. These might have been created over some longer period of time - say 1-2 years using different versions of the API toolbox. At some point your scripts might not be compatible with newer versions of the API toolbox, but updating all of them takes too much effort and is not worth it as you know that your scripts worked just fine when you created them. A solution would now be to switch between versions of the API toolbox to not have to update the scripts to newer versions. Now you need to keep track of which version your script was using. You could keep track of this using:

* A text note in your script,
* Naming your GH script accordingly.

If this feature was implemented you would not have to keep track manually.

This is the one use case I can think of.

**Comment by xRadne** [comment]
Ok sure! 👨‍💻

**Comment by xRadne** [comment]
Fixed it! 
![image](https://user-images.githubusercontent.com/32938476/183058261-90acdcd9-4ab4-4e34-b512-0d68a2ca5a01.png)
![image](https://user-images.githubusercontent.com/32938476/183058340-783f9767-efaf-47f1-b872-0c7e1c8689db.png)
